platform/android/export/export.cpp

```diff
@@ -1341,6 +1356,49 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			return str;
 		}
 	}
+
+	// Creates strings.xml files inside the gradle project for different locales.
+	void _create_project_name_strings_files(const Ref<EditorExportPreset> &p_preset) {
+		// TODO: should we use "package/name" or "application/config/name"?
+		String package_name = p_preset->get("package/name");
+		String project_name = get_project_name(package_name);
+		const String base_xml_string =
+				"<?xml version=\"1.0\" encoding=\"utf-8\"?>\n"
+				"<!--WARNING: THIS FILE WILL BE OVERWRITTEN AT BUILD TIME-->\n"
+				"<resources>\n"
+				"\t<string name=\"godot_project_name_string\">%s</string>\n"
+				"</resources>\n";
+		// Stores the string into the defualt values directory.
+		String processed_default_xml_string = vformat(base_xml_string, project_name.xml_escape(true));
+		store_string_at_path("res://android/build/res/values/godot_project_name_string.xml", processed_default_xml_string);
+
+		// Searches the Gradle project res/ directory to find all supported locales
+		DirAccessRef da = DirAccess::open("res://android/build/res");
+		if (da) {
+			da->list_dir_begin();
+			while (true) {
+				String file = da->get_next();
+				if (file == "") {
+					break;
+				}
+				if (file.begins_with("values-")) {
```

Comment:
To reduce nesting:

```
if (!file.begins_with("values-")) {
  break;
}
```

Reply:
I don't think this achieves the same behavior. For example, the `res/` directory will have directories such as `mipmap`. However, those `mipmap` directories may be given to us by `da->get_next()` before we are done going through all the `values-xx` files. In that scenario, we would not want to break out of the loop.

Reply:
Oops, my bad - I had meant to suggest `continue;` here.

Reply:
But then the loop will never end.